core/tests/protractor_utils/ExplorationEditorSettingsTab.js

```diff
@@ -161,25 +169,27 @@ var ExplorationEditorSettingsTab = function() {
   };
 
   this.expectCategoryToBe = async function(category) {
+    var containerLocator = by.css('.e2e-test-exploration-category-dropdown');
     await waitFor.presenceOf(
-      explorationCategoryInput,
+      explorationCategory,
       'Exploration category input takes too long to be visible.');
-    expect(await explorationCategoryInput.$('option:checked').getText()).
+    expect(await element(containerLocator).getText()).
       toEqual(category);
```

Comment:
Why not just create the full element with `containerLocator` after the `waitFor.presenceOf` instead of splitting it locator definition and the element selection across two lines like this?

Reply:
done